src/PrettierEditProvider.ts

```diff
@@ -107,9 +107,28 @@ class PrettierEditProvider implements
         token: CancellationToken
     ): TextEdit[] {
         try {
+            const text = document.getText();
+            
+            if (document.fileName.slice(0, -4) === '.vue') {
```

Comment:
The file extension doesn't work when file is unsaved.
Should be `document.languageId === 'vue' `

I would also extract this next part into a function.